build: downgrade to go 1.20

giuseppe opened the pull request:
go 1.21 breaks the Podman CI since it is not available on Fedora 38.

<!--
Thanks for sending a pull request!

Please make sure you've read and understood our contributing guidelines
(https://github.com/containers/buildah/blob/main/CONTRIBUTING.md) as well as ensuring
that all your commits are signed with `git commit -s`.
-->

#### What type of PR is this?

<!--
Please label this pull request according to what type of issue you are
addressing, especially if this is a release targeted pull request.

Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### What this PR does / why we need it:

#### How to verify it

#### Which issue(s) this PR fixes:

<!--
Automatically closes linked issue when PR is merged.
Uncomment the following comment block and include the issue
number or None on one line.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`, or `None`.
-->

<!--
Fixes #
or
None
-->

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes please follow the kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

nalind commented:
/lgtm
/hold

mtrmac reviewed (commented):
https://github.com/containers/buildah/pull/5139/files also changed the `vendor-in-container` rule.

I’m not sure this is completely necessary but the consistency is attractive.

giuseppe commented:
> https://github.com/containers/buildah/pull/5139/files also changed the `vendor-in-container` rule.
> 
> I’m not sure this is completely necessary but the consistency is attractive.

good point, fixed now

flouthoc reviewed (approved):
LGTM
/approve

flouthoc commented:
/lgtm
/hold

flouthoc commented:
Restarted flaking test, @giuseppe I think PR needs rebase. ( otherwise merge bot will block )

nalind commented:
/lgtm

mtrmac commented:
/unhold
